### PR TITLE
Fix WebGLRenderbuffer memory leak from MSAA RenderTargetTextures

### PR DESCRIPTION
**Pull request by rapid-images-tore-levenstam**
Forum thread with more info:
https://forum.babylonjs.com/t/babylon-leaks-msaa-renderbuffers-on-resize-dispose-with-webgl/51228

The short version is that renderbuffers created by MSAA RenderTargetTextures were never deleted, causing a (sometimes big) memory leak. In our project it could be several gigabytes of leaked memory in a few seconds.
The proposed solution here is that thinEngine should use the `webGLHardwareTexture`'s function `release` to let it delete its resources, instead of directly deleting (only) the WebGLTexture from the context directly.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by rapid-images-tore-levenstam** [comment]
I want to note here that this fix seemed to work in our project which runs an earlier version of Babylon (6.31.0). 
I'm not sure if this change has other implications, but I couldn't find any direct issues from my limited testing.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15184/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15184/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15184/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15184/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15184/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15184/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15184/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15184/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15184/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15184/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15184/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15184/merge/testResults/webgpuplaywright/index.html

**Comment by rapid-images-tore-levenstam** [comment]
Not sure why the formatting fails, or the WebGL1 visualization tests fail. Any ideas?

**Comment by RaananW** [comment]
> Not sure why the formatting fails, or the WebGL1 visualization tests fail. Any ideas?

Formatting fails because something is off :-)

Run `npm run format:fix`, it will fix everything for you. Probably some whitespace issue.

About WebGL1 - I am running the tests again just to be sure it is related to this PR.

File: packages/dev/core/src/Engines/thinEngine.ts [hunk 1]
```diff
@@ -3529,10 +3529,8 @@ export class ThinEngine extends AbstractEngine {
         }
     }
 
-    protected _deleteTexture(texture: Nullable<WebGLTexture>): void {
-        if (texture) {
-            this._gl.deleteTexture(texture);
-        }
+    protected _deleteTexture(texture: Nullable<WebGLHardwareTexture>): void {
```
Comment (RaananW) [inline packages/dev/core/src/Engines/thinEngine.ts:3532]: Please make sure to change NativeEngine as well. CC @bghgary

Comment (RaananW) [reply packages/dev/core/src/Engines/thinEngine.ts:3532]: Just a thought - it seems like this change makes _deleteTexture redundant, which makes sense to me (though I will wait for other reviewers). It is possible to remove `_deleteTexture`, and only use the texture's release function.

**Comment by bjsplat** [comment]
Visualization tests for WebGL 1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15184/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by rapid-images-tore-levenstam** [comment]
> > Not sure why the formatting fails, or the WebGL1 visualization tests fail. Any ideas?
> 
> Formatting fails because something is off :-)
> 
> Run `npm run format:fix`, it will fix everything for you. Probably some whitespace issue.
> 
> About WebGL1 - I am running the tests again just to be sure it is related to this PR.

Alright, thanks for the tip, trying it :)

File: packages/dev/core/src/Engines/thinEngine.ts (hunk 1 above)
Comment (rapid-images-tore-levenstam) [reply packages/dev/core/src/Engines/thinEngine.ts:3532]: Indeed, it looks like _deleteTexture can be deleted. 

I haven't worked with the NativeEngine, and I can't tell if I can delete the `_deleteTexture` from there as well. 
`INativeEngine` says the function `deleteTexture` is there, which is used by the `nativeEngine`'s `_deleteTexture`. But I can't find any implementers of the function anywhere. Thus I can't say if it's safe to just remove `_deleteTexture` from the `nativeEngine`.
I'm also unsure what type it expects. The nativeEngine's version takes a WebGLTexture, but uses it as an Uint32Array.

**Comment by bjsplat** [comment]
Visualization tests for WebGL 1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15184/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

File: packages/dev/core/src/Engines/thinEngine.ts (hunk 1 above)
Comment (RaananW) [reply packages/dev/core/src/Engines/thinEngine.ts:3532]: let's keep it this way for now. I'll wait for the native team to comment here.

**Review by deltakosh — APPROVED** [review]
Thanks for the fix!

**Review by RaananW — APPROVED** [review]

**Pull request merged**